server/src/main/java/timely/api/model/Metric.java

```diff
@@ -34,10 +34,23 @@
 @WebSocket(operation = "put")
 public class Metric implements TcpRequest, HttpPostRequest, WebSocketRequest {
 
+    private static final ThreadLocal<ByteBuffer> WRITE_BUFFER = new ThreadLocal<ByteBuffer>() {
+
+        @Override
+        protected ByteBuffer initialValue() {
+            return ByteBuffer.allocate(Double.BYTES);
+        }
+    };
+    private static final ThreadLocal<ByteBuffer> READ_BUFFER = new ThreadLocal<ByteBuffer>() {
```

Comment:
Thinking again where read and write occur, you might be right, I might be able to consolidate these.